### DISCUSSION TITLE
[HORIZON 4.3.3] Redis connection [horizon] not configured.

### DISCUSSION THREAD

**Discussion by Jonathanm10** [laravel/framework#37998] [Q&A]
Hello,

I have some trouble understanding something about Horizon. 

I have a Docker setup with basically three images, one with apache2, one with mysql and one with redis.

My `queue.php`, `database.php` and `horizon.php` have all the defaults, just the REDIS_HOST is set to my docker redis host.

The thing is that when trying to connect to horizon (`/horizon/api/masters`), I always get the error mentioned in the title. When looking up the code, I see that for instance `RedisMasterSupervisorRepository.php`, it forces the connection `horizon`:
```
/**
 * Get the Redis connection instance.
 *
 * @return \Illuminate\Redis\Connections\Connection
 */
protected function connection()
{
    return $this->redis->connection('horizon');
}
```

Does it mean I should configure a horizon redis database in `database.php`? It doesn't say so in any documentation. Though I tried but unsuccessfully. The error disappear, but the connection stays "inactive".
If I force the connection in `RedisManager.php` to be `default`, I have everything working.

Does any of you would know why is wrong with my setup?

Thanks a lot.

**Comment by nolanpro** [comment]
For anyone else debugging this, the horizon connection is created dynamically by copying the redis configuration set for `use`: https://github.com/laravel/horizon/blob/34ba24a8054700b20ccfe71a5dc09ffe5df74cb7/src/Horizon.php#L111

**Comment by murilolivorato** [comment]
I did a configuration , try this -
https://medium.com/@murilolivorato/mastering-laravel-horizon-the-ultimate-guide-to-queue-management-and-monitoring-a0f78784ce01
